### DISCUSSION TITLE
how to determine and adjust the COM port on windows for esp32 nodeMCU board?

### DISCUSSION THREAD

**Discussion by BerndMueller** [tbnobody/OpenDTU#321] [Q&A] [answered]
Hi,
how to determine and adjust the correct COM port in the file platformio.ini? (upload_port and monitor_port)
I bought ESP32 NodeMCU Development Board from AZ-Delivery and when I connect it with a USB cable to my
Surface Laptop 4 running win10-x64 in device-manager, it shows up under other devices as "CP2102 USB to UART Bridge Controller"
but marked with a yellow triangle warning with an exclamation mark and in the device status, it says that the driver is not installed (Code 28). I got the same on another Lenovo laptop with win7-x64.
Do I need a driver for this? Etc.? How/where to get it?
Any ideas, thoughts, or advice are highly appreciated.
Thank you,
Bernd

**Comment by BerndMueller** [comment, marked as answer]
Sorry, I could solve it in the meantime. I need a proper driver for this installed. [See here](https://blog.darrenjrobinson.com/esp32-com-port-cp2102-usb-to-uart-bridge-controller/)
After that, it is now showing up correctly as COM3 :)

**Reply by Pvler** [reply]
Hi,
I have a similar Problem. I am using an old Lenovo Laptop with Win10 and after conecting the ESP32 board nothing happens
in the device manager. I also tried to install the silicon labs quad driver but nothing of them worked so far.
The auto-detect mode in the platformio.ini file found the COM15 but then an fatal error occured:
"A fatal error occurred: Could not open COM15, the port doesn't exist"

Any ideas ? Help ?
Thank you
Daniel
